### PR TITLE
Make projectPath group-writable

### LINKED ISSUE #355: Different run-image and build-image users causing runtime failure
## What happened?

[This RFC](https://github.com/paketo-buildpacks/rfcs/pull/135/files) outlines how runtime modifications should be avoided. This can be achieved by having separate users for the run-time and build-time images.

* What were you attempting to do?
Build and run my application's image, with different run-time and build-time user ids.

* What did you expect to happen?
Build and run the my application successfully.

* What was the actual behavior? Please provide log output, if possible.
As runtime modifications are made through the `setup-symlinks` script, I am unable to run my applications image. Instead we get the following error:

```
2022/04/20 16:45:46 unlinkat /workspace/node_modules: operation not permitted
```

## Build Configuration

* What platform (`pack`, `kpack`, `tekton` buildpacks plugin, etc.) are you
  using? Please include a version.

0.24.1

* Reproducing
The issue can be reproduced here: https://github.com/g-suraj/npm_uid_issue
`./make.sh` and then run then `docker run test_image`
  

## Checklist

* [ ] I have included log output.
* [ ] The log output includes an error message.
* [ ] I have included steps for reproduction.

### PR DESCRIPTION
**Pull request by thitch97**
## Summary
Makes the projectPath group-writable at the end of the build phase.

## Use Cases
Resolves #355 

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by thitch97** [comment]
I have been able to do integration testing on this locally but have opted to hold off on including integration tests here since there are not yet any official builders whose stacks use different build and run user ids. Separate issue & PR will follow.

**Comment by g-suraj** [comment]
Anything in particular holding this PR from being approved/merged? Seems like it's good to go

**Comment by thitch97** [comment]
> Anything in particular holding this PR from being approved/merged? Seems like it's good to go

@g-suraj A couple things, yes. My fellow subteam maintainer is out for another week or so and this will require their approval. There were also concerns over whether this approach goes against the intentions of the [Secure Runtime Environments RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0045-user-ids.md) in its permissiveness but that isn't necessarily an outright blocker.

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**
